### PR TITLE
Remove build tool from archetype template

### PR DESCRIPTION
**Pull request by srnagar**
`azure-maven-plugin` is not yet released to Maven Central. So, this PR removes the plugin from the `<build>` section of the azure-sdk-archetype generated projects. This will be added back to the archetype when the build plugin is released.

### CODE REVIEW

**Review by JonathanGiles — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 1]
```diff
@@ -44,6 +44,9 @@ parameters:
 - name: MatrixReplace
   type: object
   default: []
+- name: SkipAggregateReports
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml:47]: Where is this used?

Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml:47]: Looking at https://github.com/Azure/azure-sdk-for-java/search?q=SkipAggregateReports&type= perhaps you missed this in your last PR. However shouldn't this still be passed down somewhere?

Comment (srnagar) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml:47]: Yeah, I had missed passing this param down to the test yml. Fixed now.

**Pull request merged**